examples/kmeans/kmeans.py

```diff
@@ -0,0 +1,136 @@
+                int dist_ind1[] = {i / dim, 0};
+                int dist_ind2[] = {i / dim, 1};
+                double diff1 = centers[cent_ind1] - data;
+                double diff2 = centers[cent_ind2] - data;
+                atomicAdd(&dist[dist_ind1], diff1 * diff1);
+                atomicAdd(&dist[dist_ind2], diff2 * diff2);
+                ''',
+                'calc_distances'
+            )(X, centers, data_dim, distances)
+
+        new_pred = xp.argmin(distances, axis=1).astype(np.int32)
+        if xp.all(new_pred == pred):
+            break
+        pred = new_pred
+
+        # calculate centers
+        if not elem or xp == np:
+            centers = xp.stack([X[pred == i].mean(axis=0)
+                                for i in six.moves.range(n_clusters)])
+        else:
+            centers = xp.zeros((n_clusters, data_dim),
+                               dtype=np.float32)
+            group = xp.zeros(n_clusters, dtype=np.float32)
+            label = pred[:, None]
+            cupy.ElementwiseKernel(
+                'S data, T label, int32 dim', 'raw S centers, raw S group',
+                '''
+                int cent_ind[] = {label, i % dim};
+                atomicAdd(&centers[cent_ind], data);
+                atomicAdd(&group[label], 1);
+                ''',
+                'calc_center'
+            )(X, label, data_dim, centers, group)
+            group /= data_dim
+            centers /= group[:, None]
+
+    return centers, pred
+
+
+def draw(X, n_clusters, centers, pred, output):
+    xp = cupy.get_array_module(X)
+    for i in six.moves.range(n_clusters):
+        labels = X[pred == i]
+        if xp == cupy:
+            labels = labels.get()
+        plt.scatter(labels[:, 0], labels[:, 1], color=np.random.rand(3, 1))
+    if xp == cupy:
+        centers = centers.get()
+    plt.scatter(centers[:, 0], centers[:, 1], s=120, marker='s',
+                facecolors='y', edgecolors='k')
+    plt.savefig(output + '.png')
+
+
+def run(gpuid, n_clusters, max_iter, elem, output):
+    samples = np.random.randn(5000000, 2).astype(np.float32)
+    X_train = np.r_[samples + 1, samples - 1]
+    repeat = 1
+
+    with timer(' CPU '):
+        for i in range(repeat):
+            centers, pred = use_custom_kernel(X_train, n_clusters,
+                                              max_iter, elem)
+
+    with cupy.cuda.Device(gpuid):
+        X_train = cupy.asarray(X_train)
+        with timer(' GPU '):
+            for i in range(repeat):
+                centers, pred = use_custom_kernel(X_train, n_clusters,
+                                                  max_iter, elem)
+        if output is not None:
+            index = np.random.choice(10000000, 300, replace=False)
+            draw(X_train[index], n_clusters, centers, pred[index], output)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--gpu-id', '-g', default=0, type=int, dest='gpuid',
```

Comment:
When `dest` is omitted, its result is stored to `gpu_id`, and it is enough and simpler.